match_with_noaa_stations.py

```diff
@@ -77,61 +117,66 @@ def retrieve_noaa_data(token):
     df = pd.DataFrame.from_dict(results)
     return df
 
-
 def find_closest_noaa_station(noaa_stations, row):
     """Find the closest station given a row from the circle data
-
     :param list noaa_stations:
         A list of dictionaries representing the NOAA stations and their
         coordinates
     :param pandas.core.series.Series row:
         a row from a pandas DataFrame
-
     :return: the closest NOAA station and their coordinates
     :rtype: dict
     """
     lat_lng_pair = (row['lat'], row['lon'])
+    count_year = pd.to_datetime(row['count_date']).year
     shortest = pd.np.Inf
 
     for station in noaa_stations:
         calc_distance = haversine_formula(station['coordinates'], lat_lng_pair)
-        if  calc_distance < shortest:
+        #if  calc_distance < shortest:
+        if  calc_distance < shortest and count_year >= station['firstyear'] and count_year <= station['lastyear']:
             closest_noaa = {
                 'circle_name': row['circle_name'],
+                'circle_num': row['Unnamed__0'],
```

Comment:
What is this here? If this is inherent with the flat file, we should change the column name.